### PR TITLE
fix: CA1308

### PR DESCRIPTION
**Pull request by daniel-akili**
Related to #202

### CODE REVIEW

**Review by RushabhBhansali — COMMENTED** [review]
I only see one file changed in the PR. Did your changes got reverted in later commits ?

File: src/Microsoft.ComponentDetection.Detectors/rust/Contracts/CargoPackage.cs [hunk 1]
```diff
@@ -39,7 +39,9 @@ public override int GetHashCode()
         {
             return HashCode.Combine(
                 EqualityComparer<string>.Default.GetHashCode(this.name),
+#pragma warning disable CA1308
```
Comment (RushabhBhansali) [inline src/Microsoft.ComponentDetection.Detectors/rust/Contracts/CargoPackage.cs:42]: this suppression can be moved to method level.

File: src/Microsoft.ComponentDetection.Detectors/pip/PythonVersion.cs [hunk 2]
```diff
@@ -174,8 +174,10 @@ private static int ComparePreRelease(PythonVersion a, PythonVersion b)
                 return -1;
             }
 
+#pragma warning disable CA1308
```
Comment (RushabhBhansali) [inline src/Microsoft.ComponentDetection.Detectors/pip/PythonVersion.cs:177]: this suppression can be moved to method level.

File: src/Microsoft.ComponentDetection.Contracts/TypedComponent/PipComponent.cs [hunk 3]
```diff
@@ -21,7 +21,9 @@ public PipComponent(string name, string version)
 
         public override ComponentType Type => ComponentType.Pip;
 
+#pragma warning disable CA1308
```
Comment (RushabhBhansali) [inline src/Microsoft.ComponentDetection.Contracts/TypedComponent/PipComponent.cs:24]: use `SuppressMessage` attribute instead.

File: src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs [hunk 4]
```diff
@@ -75,7 +75,9 @@ protected override async Task OnFileFound(ProcessRequest processRequest, IDictio
                 }
                 else
                 {
+#pragma warning disable CA1308
```
Comment (RushabhBhansali) [inline src/Microsoft.ComponentDetection.Detectors/go/GoComponentDetector.cs:78]: no need to suppress here,  instead  we can use .toUpperInvariant` and change `.mod` -> `.MOD` and `.sum` -> `.SUM` in the switch statement below.

**Review by RushabhBhansali — COMMENTED** [review]
Provided

File: src/Microsoft.ComponentDetection.Detectors/cocoapods/PodComponentDetector.cs [hunk 5]
```diff
@@ -404,7 +404,9 @@ public string GetSpecRepositoryOfSpec(string specName)
                     {
                         // CocoaPods specs are stored in a git repo but depending on settings/CocoaPods version
                         // the repo is shown differently in the Podfile.lock
+#pragma warning disable CA1308
                         return repository.Key.ToLowerInvariant() switch
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Detectors/cocoapods/PodComponentDetector.cs:408]: I think you could use `ToUpperInvariant()` here as well, and change the switch case from `trunk` to `TRUNK`.

**Review by RushabhBhansali — APPROVED** [review]

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**